### PR TITLE
Fabric and Configurate fixes

### LINKED ISSUE #17: Doesn't work for me
Doesn't work, mc crashes when trying to launch, the only other mods I have is fabric and fabric api. I tried with both older and newer fabric versions.

### PR DESCRIPTION
**Pull request by NotRyken**
Fixes #17 (Configurate dependency notation causes crash on load) and improves fabric.mod.json

### CODE REVIEW

**Comment by Scubakay** [comment]
Thank you! I'll check this when I have time

File: build.gradle [hunk 1]
```diff
@@ -31,7 +31,11 @@ dependencies {
     minecraft "com.mojang:minecraft:${project.minecraft_version}"
     mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
     modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
-    implementation("org.spongepowered:configurate-hocon:4.1.2")
+
+    modApi(include("org.spongepowered:configurate-core:4.1.2"))
+    modApi(include("org.spongepowered:configurate-hocon:4.1.2"))
+    include("io.leangen.geantyref:geantyref:1.3.13")
+    include("com.typesafe:config:1.4.2")
```
Comment (Scubakay) [inline build.gradle:38]: I understand this is to make hocon typesafe?

File: build.gradle [hunk 2]
```diff
@@ -31,7 +31,11 @@ dependencies {
     minecraft "com.mojang:minecraft:${project.minecraft_version}"
     mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
     modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
-    implementation("org.spongepowered:configurate-hocon:4.1.2")
+
+    modApi(include("org.spongepowered:configurate-core:4.1.2"))
+    modApi(include("org.spongepowered:configurate-hocon:4.1.2"))
+    include("io.leangen.geantyref:geantyref:1.3.13")
```
Comment (Scubakay) [inline build.gradle:37]: What is this for?

File: build.gradle [hunk 3]
```diff
@@ -31,7 +31,11 @@ dependencies {
     minecraft "com.mojang:minecraft:${project.minecraft_version}"
     mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
     modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
-    implementation("org.spongepowered:configurate-hocon:4.1.2")
+
+    modApi(include("org.spongepowered:configurate-core:4.1.2"))
```
Comment (Scubakay) [inline build.gradle:35]: I think we should move the versions to gradle.properties, but I can do that myself before creating a new release

File: src/main/resources/fabric.mod.json [hunk 4]
```diff
@@ -20,8 +27,8 @@
     "autorelog.mixins.json"
   ],
   "depends": {
-    "fabricloader": ">=${loader_version}",
-    "fabric": "*",
+    "fabricloader": ">=0.12.23",
```
Comment (Scubakay) [inline src/main/resources/fabric.mod.json:30]: This version should move to gradle.properties

**Comment by Scubakay** [comment]
Thanks for the PR! Added some notes and questions.

**Review by Scubakay — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (NotRyken) [reply build.gradle:38]:
```
java.lang.RuntimeException: Could not execute entrypoint stage 'client' due to errors, provided by 'autorelog'!
	at net.fabricmc.loader.impl.FabricLoaderImpl.lambda$invokeEntrypoints$2(FabricLoaderImpl.java:388)
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
	at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:386)
	at net.fabricmc.loader.impl.game.minecraft.Hooks.startClient(Hooks.java:53)
	at net.minecraft.class_310.<init>(class_310.java:487)
	at net.minecraft.client.main.Main.main(Main.java:223)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:87)
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:130)
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70)
Caused by: java.lang.NoClassDefFoundError: com/typesafe/config/ConfigException
	at com.scubakay.autorelog.config.Config.<init>(Config.java:47)
	at com.scubakay.autorelog.AutoRelogClient.<clinit>(AutoRelogClient.java:16)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:467)
	at net.fabricmc.loader.impl.util.DefaultLanguageAdapter.create(DefaultLanguageAdapter.java:50)
	at net.fabricmc.loader.impl.entrypoint.EntrypointStorage$NewEntry.getOrCreate(EntrypointStorage.java:117)
	at net.fabricmc.loader.impl.entrypoint.EntrypointContainerImpl.getEntrypoint(EntrypointContainerImpl.java:53)
	at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:384)
	... 9 more
Caused by: java.lang.ClassNotFoundException: com.typesafe.config.ConfigException
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:226)
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:119)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
	... 17 more
```

File: build.gradle (hunk 2 above)
Comment (NotRyken) [reply build.gradle:37]:
```
java.lang.RuntimeException: Could not execute entrypoint stage 'client' due to errors, provided by 'autorelog'!
	at net.fabricmc.loader.impl.FabricLoaderImpl.lambda$invokeEntrypoints$2(FabricLoaderImpl.java:388)
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
	at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:386)
	at net.fabricmc.loader.impl.game.minecraft.Hooks.startClient(Hooks.java:53)
	at net.minecraft.class_310.<init>(class_310.java:487)
	at net.minecraft.client.main.Main.main(Main.java:223)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:87)
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:130)
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70)
Caused by: java.lang.NoClassDefFoundError: io/leangen/geantyref/TypeToken
	at org.spongepowered.configurate.serialize.ScalarSerializer.<init>(ScalarSerializer.java:71)
	at org.spongepowered.configurate.serialize.BooleanSerializer.<init>(BooleanSerializer.java:39)
	at org.spongepowered.configurate.serialize.Scalars.<clinit>(Scalars.java:46)
	at org.spongepowered.configurate.serialize.TypeSerializerCollection.<clinit>(TypeSerializerCollection.java:50)
	at org.spongepowered.configurate.ConfigurationOptions$Lazy.<clinit>(ConfigurationOptions.java:56)
	at org.spongepowered.configurate.ConfigurationOptions.defaults(ConfigurationOptions.java:73)
	at org.spongepowered.configurate.loader.AbstractConfigurationLoader$Builder.<init>(AbstractConfigurationLoader.java:269)
	at org.spongepowered.configurate.hocon.HoconConfigurationLoader$Builder.<init>(HoconConfigurationLoader.java:96)
	at org.spongepowered.configurate.hocon.HoconConfigurationLoader.builder(HoconConfigurationLoader.java:85)
	at com.scubakay.autorelog.config.Config.<init>(Config.java:47)
	at com.scubakay.autorelog.AutoRelogClient.<clinit>(AutoRelogClient.java:16)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:467)
	at net.fabricmc.loader.impl.util.DefaultLanguageAdapter.create(DefaultLanguageAdapter.java:50)
	at net.fabricmc.loader.impl.entrypoint.EntrypointStorage$NewEntry.getOrCreate(EntrypointStorage.java:117)
	at net.fabricmc.loader.impl.entrypoint.EntrypointContainerImpl.getEntrypoint(EntrypointContainerImpl.java:53)
	at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:384)
	... 9 more
Caused by: java.lang.ClassNotFoundException: io.leangen.geantyref.TypeToken
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:226)
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:119)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
	... 26 more
```

File: build.gradle (hunk 3 above)
Comment (NotRyken) [reply build.gradle:35]: Sure, but this PR doesn't change that.

File: src/main/resources/fabric.mod.json (hunk 4 above)
Comment (NotRyken) [reply src/main/resources/fabric.mod.json:30]: Understandable.

**Comment by NotRyken** [comment]
I can't support this PR at this time.

**Comment by Scubakay** [comment]
I can't reproduce this.

**Comment by NotRyken** [comment]
I can still consistently reproduce it, and it is also the cause of the crash referenced in #17. See below for a summary of client specifications, and a crash log. I am unable to help you further with this issue.

- Launcher: Prism 8.0
- Minecraft: 1.20.4
- Fabric Loader: 0.15.6
- Fabric API: 0.95.1+1.20.4
- AutoRelog: 0.0.6+1.20.4

https://mclo.gs/g4wlHlW

**Comment by Scubakay** [comment]
Okay, thanks for the help!

**Pull request closed without merging**